shared-libs/transitions/test/unit/due_tasks.js

```diff
@@ -1779,4 +1780,193 @@ describe('due tasks', () => {
       });
     });
   });
+
+  it('change task state to clear when message generation fails and clear_failing_schedules is true', () => {
+    config.get.withArgs('clear_failing_schedules').returns(true);
+    const doc = {
+      scheduled_tasks: [{
+        due: moment().toISOString(),
+        state: 'scheduled',
+        message_key: 'k1',
+        recipient: 'patient'
+      }],
+    };
+
+    sinon.stub(schedule._lineage, 'hydrateDocs').resolves([doc]);
+    sinon.stub(request, 'get').resolves({
+      rows: [{ id: 'xyz', key: ['scheduled', Date.now()], doc }],
+    });
+
+    const messageUtils = require('@medic/message-utils');
+    // Stub the function on the actual module reference
+    sinon.stub(messageUtils, 'generate').returns([{ error: 'missing patient' }]);
+    sinon.stub(messageUtils, 'hasError').returns(true);
+
+    const setTaskState = sinon.stub(utils, 'setTaskState');
+    const saveDoc = sinon.stub(db.medic, 'put').resolves({});    
+    return schedule.execute().then(() => {
+      assert.equal(setTaskState.callCount, 1);
+      sinon.assert.calledWithMatch(
+        setTaskState,
+        sinon.match.object,
+        'clear'
+      );
+      assert.equal(saveDoc.callCount, 1);
+    });
+  });
+
+  it('dont change task state if message generation fails and clear_failing_schedules is false', () => {
+    config.get.withArgs('clear_failing_schedules').returns(false);
+    const id = 'xyz';
+    const due = moment();
+    const doc = {
+      scheduled_tasks: [
+        {
+          due: due.toISOString(),
+          state: 'scheduled',
+          message_key: 'x',
+        },
+      ],
+    };
+
+    const hydrate = sinon.stub(schedule._lineage, 'hydrateDocs').resolves([doc]);    
+    const view = sinon.stub(request, 'get').resolves({
+      rows: [
+        { id, key: ['scheduled', due.valueOf()], doc },
+      ],
+    });
+
+    // Simulate missing messages to trigger clear_failing_schedules logic    
+    const translate = sinon
+      .stub(utils, 'translate')
+      .returns('');
+    translate.hasError=true;
```

Comment:
I can't tell what the purpose of this assignment is. `translate` is a function that return an empty string.